dll/win32/shell32/folders/CDesktopFolder.cpp

```diff
@@ -281,22 +298,16 @@ HRESULT WINAPI CDesktopFolder::FinalConstruct()
 
 HRESULT CDesktopFolder::_GetSFFromPidl(LPCITEMIDLIST pidl, IShellFolder2** psf)
 {
-    WCHAR szFileName[MAX_PATH];
-
-    if (_ILIsSpecialFolder(pidl))
+    if (IsRegItem(pidl))
         return m_regFolder->QueryInterface(IID_PPV_ARG(IShellFolder2, psf));
-
-    lstrcpynW(szFileName, sPathTarget, MAX_PATH - 1);
-    PathAddBackslashW(szFileName);
-    int cLen = wcslen(szFileName);
-
-    if (!_ILSimpleGetTextW(pidl, szFileName + cLen, MAX_PATH - cLen))
-        return E_FAIL;
-
-    if (GetFileAttributes(szFileName) == INVALID_FILE_ATTRIBUTES)
-        return m_SharedDesktopFSFolder->QueryInterface(IID_PPV_ARG(IShellFolder2, psf));
+#if DBG
+    if (_ILIsDesktop(pidl))
+        FIXME("Desktop is unexpected here!\n");
     else
-        return m_DesktopFSFolder->QueryInterface(IID_PPV_ARG(IShellFolder2, psf));
+        ASSERT(!_ILIsSpecialFolder(pidl));
```

Comment:
Please enclose these branches of the `if` in `{` and `}`, otherwise (depending on how the `ASSERT` / `FIXME` is disabled, you might skip the next line instead of the macro...